src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```diff
@@ -221,6 +217,18 @@ public static void AddWebJobsScriptHost(this IServiceCollection services, IConfi
             services.AddAzureBlobStorageProvider();
         }
 
+        internal static void AddHostingConfigOptions(this IServiceCollection services, IConfiguration configuration)
+        {
+            services.ConfigureOptions<FunctionsHostingConfigOptionsSetup>();
+
+            if (configuration != null)
+            {
+                // Refresh IOptionsMonitor<FunctionsHostingConfigOptions> when the config section changes
+                var configSection = configuration.GetSection(ScriptConstants.FunctionsHostingConfigSectionName);
+                services.AddSingleton<IOptionsChangeTokenSource<FunctionsHostingConfigOptions>>(new ConfigurationChangeTokenSource<FunctionsHostingConfigOptions>(configSection));
```

Comment:
note -- the call to `Configure()` did this internally already (and we have tests ensuring that `IOptionsMonitor<>` works) but it also follws up with a call to `Bind()`, which we don't want. So separating this out and avoids the call to `Bind()`.

Reply:
If we no longer Bind, does that mean we'll lose the ability to configure hosting config via app settings? For example say we've rolled out a particular config value, but it breaks a customer resulting in a CRI. We could currently tell that customer to revert the config via app settings.

Reply:
That will work once the ordering is fixed (I'd rather do that separately). Ultimately, the settings are all fulfilled via `IConfiguration` today, just via our custom `Setup` class here: https://github.com/Azure/azure-functions-host/blob/937267a5d78b006456446be62ecb120d10c0208d/src/WebJobs.Script/Config/FunctionsHostingConfigOptionsSetup.cs#L23C1-L36C10.

That populates `Features` with `IConfiguration`. So, if a customer adds an app setting that overrides a specific value being looked for here, it'll work. You won't need setters, either -- the getter can evaluate what's in the `Features` dict and decide true/false from there.

The `Bind()` wasn't even doing anything unless the Propery name matched the ScmHostingConfig key name -- which, until you added yours, had never happened. Then I followed up with the same thing (matching names) and fell into the same issue.